### PR TITLE
refactor: make pinning instructions a fail message

### PR DESCRIPTION
**Pull request by thesayyn**
<s>BEHAVIOR CHANGE: `reproducible = True` is used to stop the execution of the repository rule. it's not the case anymore

the reason for this is that to fetch the manifest, we have to be inside a repository_rule and not a macro, if we stop the execution early in the macro then users wouldn't get the warning message. we could use `fail` instead of `print` to stop the execution. However, I think this matches the behavior of other rules like `http_archive`</s>

the reason why delta is big 
- <s>replacing usage of `sha256sum` with `coreutils`</s>
- removes `pin_tag` rule with `oci_alias` which is responsible for printing the warning.
- replaces bazel runnable target that prints the instructions with buildozer command

### CODE REVIEW

**Comment by thesayyn** [comment]
ping @alexeagle

**Review by alexeagle — COMMENTED** [review]
Could you answer "why" in the PR description?

File: oci/pull.bzl [hunk 1]
```diff
@@ -418,97 +417,86 @@ alias(
...
 
+_SINGLE_PLATFORM_IMAGE_ALIAS = """\
+alias(
+    name = "{target_name}",
+    actual = "@{original}",
+    visibility = ["//visibility:public"],
+)
+"""
+
+def _coreutils_label(rctx):
```
Comment (alexeagle) [inline oci/pull.bzl:428]: I don't understand why the delta is so large, the PR description just says you're changing an error to a warning. Is this related?

Comment (thesayyn) [reply oci/pull.bzl:428]: yes, it replaces `sha256sum` with the coreutils equivalent.

**Comment by thesayyn** [comment]
> Could you answer "why" in the PR description?

done. I could revert the `doesn't stop the execution part as well.`

File: oci/pull.bzl [hunk 2]
```diff
@@ -412,97 +411,78 @@ alias(
...
-jq(
-    name = "platforms",
-    srcs = ["manifest_list.json"],
-    filter = "[(.manifests // [])[] | .platform | .os + \\"/\\" + .architecture]",
-    # Print without newlines because it's too hard to indent that to fit under the generated
-    # starlark code below.
-    args = ["--compact-output"],
-)
+        # buildifier: disable=print
+        fail("""
```
Comment (alexeagle) [inline oci/pull.bzl:449]: I am really confused now, the PR description says we are making it a warning message, but here we `fail`.

Do you mind just making a new PR that is minimal and doesn't have things crossed out in the PR description?

Comment (thesayyn) [reply oci/pull.bzl:449]: I made it fail to keep the changes to the PR minimal.
